pkg/apis/networking/validation/validation.go

```diff
@@ -206,27 +239,29 @@ func validateIngressTLS(spec *networking.IngressSpec, fldPath *field.Path) field
 }
 
 // ValidateIngressSpec tests if required fields in the IngressSpec are set.
-func ValidateIngressSpec(spec *networking.IngressSpec, fldPath *field.Path) field.ErrorList {
+func ValidateIngressSpec(spec *networking.IngressSpec, fldPath *field.Path, requestGV schema.GroupVersion) field.ErrorList {
 	allErrs := field.ErrorList{}
-	// TODO: Is a default backend mandatory?
-	if spec.Backend != nil {
-		allErrs = append(allErrs, validateIngressBackend(spec.Backend, fldPath.Child("backend"))...)
+	if spec.DefaultBackend != nil {
```

Comment:
I think the logic will be clearer w/out the else if:

```
if len(spec.Rules) == 0 && spec.DefaultBackend == nil {
   ...
}
if spec.DefaultBackend != nil {
   ...
}
if len(spec.Rules) > 0 { ...
```

Reply:
@bowei Just to confirm, you want to move the nil and 0 case as the first condition and then the nil DefaultBackend as the 2nd case? Is this to shortcut the "totally wrong" case first?

Reply:
Just updated this, thanks for the idea!